File: ios/RNOpenTokPublisherViewManager.m
#import <Foundation/Foundation.h>
#import "RNOpenTokPublisherViewManager.h"
#import "RNOpenTokPublisherView.h"

#if __has_include(<React/RCTBridge.h>)
#import <React/RCTBridge.h>
#elif __has_include("RCTBridge.h")
#import "RCTBridge.h"
#else
#import "React/RCTBridge.h"
#endif

@implementation RNOpenTokPublisherViewManager

@synthesize bridge = _bridge;

RCT_EXPORT_MODULE()

RCT_EXPORT_VIEW_PROPERTY(sessionId, NSString)

RCT_EXPORT_VIEW_PROPERTY(camera, NSInteger)

RCT_EXPORT_VIEW_PROPERTY(mute, BOOL)

RCT_EXPORT_VIEW_PROPERTY(video, BOOL)

RCT_EXPORT_VIEW_PROPERTY(screenCapture, BOOL)

RCT_EXPORT_VIEW_PROPERTY(screenCaptureSettings, NSDictionary*)

- (UIView *)view {
    return [[RNOpenTokPublisherView alloc] initWithUIManager:_bridge.uiManager];
}

@end

File: android/src/main/java/com/rnopentok/RNOpenTokPublisherViewManager.java
package com.rnopentok;

import android.support.annotation.Nullable;

import com.facebook.react.uimanager.ThemedReactContext;
import com.facebook.react.uimanager.annotations.ReactProp;
import com.facebook.react.bridge.ReadableMap;


public class RNOpenTokPublisherViewManager extends RNOpenTokViewManager<RNOpenTokPublisherView> {
    @Override
    public String getName() {
        return "RNOpenTokPublisherView";
    }

    @Override
    protected RNOpenTokPublisherView createViewInstance(ThemedReactContext reactContext) {
        return new RNOpenTokPublisherView(reactContext);
    }

    @ReactProp(name = "mute")
    public void setMute(RNOpenTokPublisherView view, Boolean mute) {
        view.setAudio(!mute);
    }

    @ReactProp(name = "video")
    public void setVideo(RNOpenTokPublisherView view, Boolean video) {
        view.setVideo(video);
    }

    @ReactProp(name = "camera")
    public void setCamera(RNOpenTokPublisherView view, Integer camera) {
        if (camera != 0) {
            view.cycleCamera();
        }
    }

    @ReactProp(name = "cameraDirection")
    public void setCameraDirection(RNOpenTokPublisherView view, String cameraDirection) {
        view.setCameraDirection(RNOpenTokPublisherView.CameraDirection.valueOf(cameraDirection.toUpperCase()));
    }

    @ReactProp(name = "screenCapture")
    public void setScreenCapture(RNOpenTokPublisherView view, Boolean screenCapture) {
        view.setScreenCapture(screenCapture);
    }

    @ReactProp(name = "screenCaptureSettings")
    public void setScreenCaptureSettings(RNOpenTokPublisherView view, @Nullable ReadableMap screenCaptureSettings) {
        view.setScreenCaptureSettings(screenCaptureSettings);
    }
}

